lib/rspec/mocks/error_generator.rb

```diff
@@ -62,6 +62,18 @@ def raise_unexpected_message_args_error(expectation, *args)
       end
 
       # @private
+      def raise_unexpected_message_with_args_error(expectation, *args)
+        expected_args = format_args(*expectation.expected_args)
+        actual_args = format_unexpected_with_args(*args)
+        diff = diff_message(expectation.expected_args, args)
+
+        message = default_error_message(expectation, expected_args, actual_args)
+        message << "\nDiff:#{diff}" unless diff.empty?
+
+        __raise message
+      end
```

Comment:
Also, if `args` is an array of argument-arrays (e.g. one argument array per method call), as I understand it, I think a different name would be appropriate, something like `args_for_multiple_calls` or something similar.